cache/models/samples.js

```diff
@@ -586,15 +616,31 @@ module.exports = {
    * response. Note: Message body and message code will be updated if provided,
    * else no fields for message body/message code in redis object.
    * @param  {Object} params - Request parameters
+   * @param {String} userName - The user performing the write operation
    * @returns {Promise} - Resolves to a sample object
    */
-  patchSample(params) {
+  patchSample(params, userName) {
     const sampleName = params.key.value;
+    const aspectName = sampleName.split('|')[ONE];
```

Comment:
Is it possible to arrive at this point in the code with an invalid sampleName, e.g. a sampleName with no "|" character, or other ?